### PR TITLE
[ROCM] rccl kernel traces included in gpu profiler

### PR DESCRIPTION
**Pull request by zahiqbal**
Adding HIP_API_ID_hipExtLaunchKernel in xla gpu's rocm-tracer to captures rccl kernel traces.

### CODE REVIEW

**Comment by i-chaochen** [comment]
please review this PR @xla-rotation

**Review by ddunl — APPROVED** [review]

File: xla/backends/profiler/gpu/rocm_tracer.cc [hunk 1]
```diff
@@ -508,6 +512,24 @@ void RocmApiCallbackImpl::AddKernelEventUponApiExit(uint32_t cbid,
...
+          data->args.hipExtLaunchKernel.sharedMemBytes;
+      event.kernel_info.block_x = data->args.hipExtLaunchKernel.dimBlocks.x;
+      event.kernel_info.block_y = data->args.hipExtLaunchKernel.dimBlocks.y;
+      event.kernel_info.block_z = data->args.hipExtLaunchKernel.dimBlocks.z;
+      event.kernel_info.grid_x = data->args.hipExtLaunchKernel.numBlocks.x;
+      event.kernel_info.grid_y = data->args.hipExtLaunchKernel.numBlocks.y;
+      event.kernel_info.grid_z = data->args.hipExtLaunchKernel.numBlocks.z;
+      event.kernel_info.func_ptr = (void*)func_addr;
+      event.device_id = hipGetStreamDeviceId(stream);
+
```
Comment (ddunl) [inline xla/backends/profiler/gpu/rocm_tracer.cc:531]: Could you delete this line? I'd expect our autoformat-on-import to catch this but it doesn't for some reason.

Comment (akuegel) [reply xla/backends/profiler/gpu/rocm_tracer.cc:531]: Note that this is blocking the merge.

Comment (zahiqbal) [reply xla/backends/profiler/gpu/rocm_tracer.cc:531]: the same code is at line 511-513.  i wonder why the issue is not raised over there..
let me remove constantness and see how it comes up...
event.kernel_info.func_ptr = const_cast<void*>(func_addr);

Comment (zahiqbal) [reply xla/backends/profiler/gpu/rocm_tracer.cc:531]: event.device_id is vital here to place the kernel at right device, it can't be ignored.

Comment (akuegel) [reply xla/backends/profiler/gpu/rocm_tracer.cc:531]: Sorry, there was a misunderstanding here. The issue is the extra blank line, not the line with event.device_id

Comment (zahiqbal) [reply xla/backends/profiler/gpu/rocm_tracer.cc]: deleted ....

**Review by akuegel — APPROVED** [review]

**Pull request closed without merging**